weave/tests/trace/test_trace_settings.py

```diff
@@ -90,3 +93,53 @@ def test_print_call_link_env(client):
 
     output = captured_stdout.getvalue()
     assert TRACE_CALL_EMOJI in output
+
+
+def test_should_capture_code_setting(client):
+    parse_and_apply_settings(UserSettings(capture_code=False))
+
+    @weave.op
+    def test_func():
+        return 1
+
+    ref = weave.publish(test_func)
+    test_func2 = ref.get()
+    code2 = test_func2.art.path_contents["obj.py"].decode()
+    assert "Code-capture disabled" in code2
+
+    parse_and_apply_settings(UserSettings(capture_code=True))
+
+    # TODO: Not safe to change capture_code setting mid-script because the op's ref
+    # does not know about the setting change.
```

Comment:
is there a simple way around this?

Currently you have this problem:
```py
def func(): ...

os.environ["WEAVE_CODE_CAPTURE"] = False
ref = weave.publish(func)

os.environ["WEAVE_CODE_CAPTURE"] = True
ref2 = weave.publish(func)

assert ref == ref2  # True
```

Which happens because the code function has not changed

Reply:
I think this is OK behaviour as long as we document it.  I think it would be rare to flip between capturing and not capturing code mid-script